apps/typeform/frontend/src/utils.ts

```diff
@@ -0,0 +1,74 @@
+import get from 'lodash/get';
+import {
+  ContentType,
+  CompatibleFields,
+  Field,
+  EditorInterface,
+  SelectedFields,
+  TypeFormParameters
+} from './typings';
+
+function isCompatibleField(field: Field): boolean {
+  return field.type === 'Symbol';
+}
+
+export function getCompatibleFields(contentTypes: ContentType[]): CompatibleFields {
+  return contentTypes.reduce((acc, ct) => {
+    return {
+      ...acc,
+      [ct.sys.id]: (ct.fields || []).filter(isCompatibleField)
+    };
+  }, {});
+}
+
+export function editorInterfacesToSelectedFields(
+  eis: EditorInterface[],
+  appId?: string
+): SelectedFields {
+  return eis.reduce((acc, ei) => {
+    const ctId = get(ei, ['sys', 'contentType', 'sys', 'id']);
+    const fieldIds = get(ei, ['controls'], [])
+      .filter(control => control.widgetNamespace === 'app' && control.widgetId === appId)
+      .map(control => control.fieldId)
+      .filter(fieldId => typeof fieldId === 'string' && fieldId.length > 0);
+
+    if (ctId && fieldIds.length > 0) {
+      return { ...acc, [ctId]: fieldIds };
+    } else {
+      return acc;
+    }
+  }, {});
+}
+
+export function selectedFieldsToTargetState(
+  contentTypes: ContentType[],
+  selectedFields: SelectedFields
+) {
+  return {
+    EditorInterface: contentTypes.reduce((acc, ct) => {
+      const { id } = ct.sys;
+      const fields = selectedFields[id] || [];
+      const targetState =
+        fields.length > 0 ? { controls: fields.map(fieldId => ({ fieldId })) } : {};
+
+      return { ...acc, [id]: targetState };
+    }, {})
+  };
+}
+
+export const isUserAuthenticated = () => {
+  return window.localStorage.getItem('token') ? true : false;
+};
+
+export function validateParameters({
+  workspaceId,
+  accessToken
+}: TypeFormParameters): string | null {
+  if (!workspaceId.length || typeof workspaceId !== 'string') {
```

Comment:
I would switch this logic otherwise you may run into a situation where trying to access the `length` prop before checking to even see if it's a string would throw the classic "cannot read property length of undefined".
```suggestion
  if (typeof workspaceId !== 'string' || !workspaceId.length) {
```